File: backend/app/routers/students.py
from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.orm import Session
from typing import List
from ..db import get_db
from .. import models, schemas

router = APIRouter(prefix="/students", tags=["students"])

@router.post("", response_model=schemas.StudentOut)
def create_student(payload: schemas.StudentCreate, db: Session = Depends(get_db)):
    s = models.Student(
        name=payload.name.strip(),
        first_name=payload.first_name.strip(),
        class_name=payload.class_name.strip(),
        # email=None
    )
    db.add(s); db.commit(); db.refresh(s)
    return s


@router.get("", response_model=List[schemas.StudentOut])
def list_students(db: Session = Depends(get_db)):
    return db.query(models.Student).order_by(models.Student.name).all()

File: frontend/src/api/students.ts
import { api } from "../api/client";

export type Student = { id: number; name: string; first_name: string; class_name: string };

export async function listStudents(): Promise<Student[]> {
  const { data } = await api.get("/students");
  return data;
}

export async function createStudent(payload: { name: string; first_name: string; class_name: string }) {
  const { data } = await api.post("/students", payload);
  return data as Student;
}
